plugins/modules/ec2_instance.py

```diff
@@ -1973,7 +1973,7 @@ def main():
         supports_check_mode=True
     )
 
-    if not module.params.get('instance_type') and not module.params.get('launch_template'):
+    if not module.params.get('instance_type') and not module.params.get('launch_template') and module.params.get('state') != 'absent':
         module.deprecate("Default value instance_type has been deprecated, in the future you must set an instance_type or a launch_template",
                          date='2023-01-01', collection_name='amazon.aws')
```

Comment:
I guess if you want, you can also move this check as it is at line 2009 (https://github.com/ansible-collections/amazon.aws/pull/719/files#diff-a6b28ca5e2f98a5237ad2e88e544b3df3de7bbc95c8dc9088cae572ff92c2f60R2009) and something like this should also work.
```suggestion
    elif not module.params.get('instance_type') and not module.params.get('launch_template'):
        module.deprecate("Default value instance_type has been deprecated, in the future you must set an instance_type or a launch_template",
                         date='2023-01-01', collection_name='amazon.aws')
```

Reply:
I had tried that but seems to change the behavior of the `if block`, probably we could keep it as is?